cogs/numb.py

```diff
@@ -0,0 +1,98 @@
+'''This is a guess the number game.'''
+import random
+from discord.ext import commands
+
+
+class NumbGuess:  # pylint: disable=too-few-public-methods
+    '''This is a number guessing game.'''
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command(name="numguess", pass_context=True)
+    async def mycom(self, context):  # pylint: disable=too-many-statements,too-many-branches
+        '''This will start a number guessing game.'''
+        guessesTaken = 0
+        await self.bot.say('How hard would you like the number game to be? (easy/mid/hard)')
+        mode = self.bot.wait_for_message(channel=context.message.channel, timeout=30)
+
+        if mode == 'easy':
+            number = random.randint(1, 10)
+
+        elif mode == 'mid':
+            number = random.randint(1, 50)
+
+        elif mode == 'hard':
+            number = random.randint(1, 100)
+
+        #easter egg in honor of rennou, 172 uses before removal, number 1 most used rencc
+        elif mode == 'no u':
+            number = random.randint(1, 420)
+            await self.bot.say('no u')
+
+        elif mode is None:
+            await self.bot.say('you think too despacito, I am going to leave now...')
+
+        else:
+            await self.bot.say('type again')
+            mode = self.bot.wait_for_message(channel=context.message.channel, timeout=30)
+
+        while True:
+            try:
+                if mode not in ["easy", "mid", "hard", "no u"]:
+                    raise ValueError
+            except ValueError:
+                await self.bot.say('type again')
```

Comment:
This code is being repeated here and above, and is difficult to follow.  Can you find a way to structure your "checking mode" logic to loop a few times before giving up?  Ultimately, we don't want infinite loops in a command.